config/routes.rb

```diff
@@ -17,6 +17,8 @@
   end
 
   resources :nodes, only: [:index, :show, :destroy] do
+    post :assign
```

Comment:
Maybe `assign_roles` with path `assign-roles`? `assign` is too generic from my POV.